src/praisonai-agents/tests/unit/agent/test_durable_run.py

```
@@ -320,6 +320,88 @@ async def test_async_wrapper_records_and_replays_result():
+    assert result["not_safely_resumable"] is True
+    assert result["error_type"] == "NotSafelyResumable"
+    journal.close()
+
+
+def test_inflight_restart_safe_tool_is_replayed_on_resume():
+    from praisonaiagents import Agent, tool
+
+    calls = {"n": 0}
+
+    @tool(restart_safe=True)
+    def fetch_report(id: int, idempotency_key: str = "") -> str:
```

Ruff (0.16.1)

[error] Function argument `id` is shadowing a Python builtin

(A002)